pyenv/action.yml

```diff
@@ -0,0 +1,80 @@
+name: Install Pyenv Python
+description: Install and activate an interpreter via Pyenv.
+inputs:
+  python-version:
+    description: The version of Python to install.
+    required: true
+  pyenv-version:
+    description: The version of Pyenv to use. Should be a tag or branch name.
+    required: false
+    default: v2.3.21
+  pyenv-root:
+    description: Location for the PYENV_ROOT directory.
+    required: false
+    default: ${{ runner.tool_cache }}/pyenv_root
+outputs:
+  venv-dir:
+    description: The absolute path of the virtual environment directory created by this action.
+    value: ${{ steps.create-venv.outputs.venv-dir }}
+runs:
+  using: composite
+  steps:
+    - name: Cache Pyenv
+      uses: actions/cache@v3
+      with:
+        path: ${{ inputs.pyenv-root }}
+        key: ${{ matrix.os }}-${{ runner.arch }}-install-pyenv-${{ inputs.pyenv-version }}-${{ inputs.python-version }}-v1
```

Comment:
I think should theoretically include the `inputs.pyenv-root` too: if someone changes it without changing the other inputs, I think the code works, but the cache will be useless (every build will be a hit, but not contain the required info, and being a hit means it won't be re-uploaded)?

Reply:
Yeah - that was never really intended there. I flailed at the lack of ability to define a constant in a composite action across steps. I'll switch to an env var export in the 1st step.

Reply:
Ok, fixed.